src/Serilog.Sinks.EventLog/LoggerConfigurationEventLogExtensions.cs

```diff
@@ -49,13 +50,14 @@ public static class LoggerConfigurationEventLogExtensions
             bool manageEventSource = false,
             string outputTemplate = DefaultOutputTemplate,
             IFormatProvider formatProvider = null,
-            LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum)
+            LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum,
+            IEventIdProvider eventIdProvider = null)
         {
             if (loggerConfiguration == null) throw new ArgumentNullException(nameof(loggerConfiguration));
 
             var formatter = new MessageTemplateTextFormatter(outputTemplate, formatProvider);
 
-            return loggerConfiguration.Sink(new EventLogSink(source, logName, formatter, machineName, manageEventSource), restrictedToMinimumLevel);
+            return loggerConfiguration.Sink(new EventLogSink(source, logName, formatter, machineName, manageEventSource, eventIdProvider), restrictedToMinimumLevel);
```

Comment:
Perhaps this is where we do the `?? new EventIdHashProvider()`, so that `EventLogSink` can require a non-null `IEventIdProvider` argument?

Reply:
I think that would be cleaner as it keeps configuration in ```LoggerConfigurationEventLogExtensions``` and it makes the ```EventLogSink``` more transparent on behavior.  My only concern is introducing a new required parameter on the existing ```EventLogSink``` constructor.  Is this something we should be concerned with?  If so, I could make this change as requested and add a constructor overload for ```EventLogSink``` to support a custom IEventIdProvider.

Let me know what you think.

Reply:
Overloaded constructor sounds like it would be a winner :+1:

Reply:
I like the idea of an overloaded constructor on `EventLogSink` to allow us to pass in a non-null `IEventIdProvider`.

Reply:
added overloaded constructor to ```EventLogSink```.  Attempting to keep default value for IEventIdProvider in one location so I've made the more generic constructor call the more complex constructor with the default value and have ```LoggerConfigurationEventLogExtensions``` smart enough to determine which constructor to utilize.